### PR TITLE
Some tweeks that hopefully are more OO

### PR DESCRIPTION
**Pull request by Freemabe**
Let me know if I'm moving in the right direction.

### CODE REVIEW

File: students/Ben_freeman/lesson09/Mailroom_OO/mailroom_OO.py [hunk 1]
```diff
@@ -25,17 +24,21 @@ def donor_donations(self):
...
+        """adds a donation to an existing donors profile"""
+        self.donor.append(new_donate)
 
 class DonorCollection(object):
     """Class to handle all of the methods that have to do with multiple donors at once"""
 
     def __init__(self, donor_list):
-        self.donors = donor_list
+        self.donors = {a.name: a.donor for a in donor_list}
+        self.donors_object = donor_list
```
Comment (natasha-aleksandrova) [inline students/Ben_freeman/lesson09/Mailroom_OO/mailroom_OO.py]: you don't need both self.donors and self.donors_object.

As per my example, just have self.donors which is a dictionary with name as a key and Donor object as a value.

`self.donors = {donor.name: donor for donor in donors_list}`

File: students/Ben_freeman/lesson09/Mailroom_OO/mailroom_OO.py [hunk 2]
```diff
@@ -25,17 +24,21 @@ def donor_donations(self):
...
+    def message(self):
         """our default thank you message"""
-        return f"\nThank you {self.name} for your donation of ${self.donor_sum:.2f}, your donations make the work "\
-            f"of the 'American society for taking donations' possible.\n\n""Sincearly,\n\n""A low paid intern\n "
+        return f"\nThank you {self.name} for your donations totaling ${self.donor_sum:.2f}, your donations make the " \
+            f" work of the 'American society for taking donations' possible.\n\n""Sincearly,\n\n""A low paid intern\n "
 
+    def update_donor(self, new_donate):
+        """adds a donation to an existing donors profile"""
+        self.donor.append(new_donate)
```
Comment (natasha-aleksandrova) [inline students/Ben_freeman/lesson09/Mailroom_OO/mailroom_OO.py]: naming is crucial 🙂 

I didn't notice it last time, but this time reading the code I was lost at what `donor` attribute is.

The attribute name actually should be `donations`. I should be able to understand attributes, functions, classes without having to fully read the code. Just the name alone should tell me a lot about what the attribute is or what the function is expected to do.

File: students/Ben_freeman/lesson09/Mailroom_OO/Cli.py [hunk 3]
```diff
@@ -29,20 +31,30 @@ def send_thanks(donor_list_variable, query="empty_1", new_donate="empty_2"):
...
+                        return print(items.message())
 
 
-def save_file_prompt(chosen_dict):
+
+def data_handler(dictionary):
+    list_of_donors = []
+    for a in dictionary:
+        list_of_donors.append(mailroom_OO.Donor(dictionary[a],a))
+    return list_of_donors
```
Comment (natasha-aleksandrova) [inline students/Ben_freeman/lesson09/Mailroom_OO/Cli.py]: great place for a list comprehension 😉 

```
return [mailroom_OO.Donor(k, v) for k, v in dictionary.items()]
```

File: students/Ben_freeman/lesson09/Mailroom_OO/mailroom_OO.py (hunk 1 above)
Comment (Freemabe) [reply students/Ben_freeman/lesson09/Mailroom_OO/mailroom_OO.py]: I'm confused how I can make me sorting_function on lines 54-59 use Donor.attribute things like you suggested last time, if I only use a dictionary that strips away (I think) the Objectness of my Donor class objects. Can I use the input from the __init__ def (specifically donor_list) in other methods in the same class without renaming them like I did before? Thank you.

Comment (Freemabe) [reply students/Ben_freeman/lesson09/Mailroom_OO/mailroom_OO.py]: Mmm, Okay so its a dictionary that identifies all of my donor objects. I think i misread what your sent in your example, so this makes sense now. Cheers

File: students/Ben_freeman/lesson09/Mailroom_OO/mailroom_OO.py (hunk 2 above)
Comment (Freemabe) [reply students/Ben_freeman/lesson09/Mailroom_OO/mailroom_OO.py]: Will do.

File: students/Ben_freeman/lesson09/Mailroom_OO/mailroom_OO.py (hunk 1 above)
Comment (natasha-aleksandrova) [reply students/Ben_freeman/lesson09/Mailroom_OO/mailroom_OO.py]: Right - your version is a dictionary with key and values as strings.

My version, keys are name (for easy lookup) and values are Donor objects for you to work with.

**Pull request merged**